hs_core/views/__init__.py

```diff
@@ -484,6 +485,11 @@ def file_download_url_mapper(request, shortkey):
     resource, _, _ = authorize(request, shortkey, needed_permission=ACTION_TO_AUTHORIZE.VIEW_RESOURCE)
     istorage = resource.get_irods_storage()
     irods_file_path = '/'.join(request.path.split('/')[2:-1])
+    listing = istorage.listdir('/'.join(irods_file_path.split('/')[0:-1]))
+    if irods_file_path.split('/')[-1] in listing[0]:
+        # it's a folder
```

Comment:
the ```split``` call are made three times with this code addition. It'd be more performant to call it one time and assign it to a variable to be used later. Also, wouldn't ```irods_file_path.split('/')[0:-1])``` just be ```irods_file_path.split('/')```? Or perhaps ```irods_file_path``` is already sufficient without another join call for calling istorage.listdir()?

Reply:
I should've done a second pass on this while developing... yes that split and join result in no change.  I'll update that.  it's also a good idea to do the split once and assign it to a variable.  I've got to update  a test, so I'll make these changes in the same commit.